plugins/guests/fedora/cap/flavor.rb

```diff
@@ -0,0 +1,25 @@
+module VagrantPlugins
+  module GuestFedora
+    module Cap
+      class Flavor
+        def self.flavor(machine)
+          # Read the version file
+          output = ""
+          machine.communicate.sudo("grep VERSION_ID /etc/os-release") do |type, data|
+            output += data if type == :stdout
+          end
+          version = output.split("=")[1].chomp!.to_i
```

Comment:
This won't work, I would rewrite this as:

```ruby
machine.communicate.sudo("grep VERSION_ID /etc/os-release") do |type, data|
  version = data..split("=")[1].chomp.to_i if type == :stdout
end
```

And then `if version && version >= 20` condition determing and returning `:"fedora_#{version}"` and ``:fedora` otherwise.